pkg/consumer/consumer.go

```diff
@@ -1,7 +1,23 @@
 package consumer
 
-import "io"
+import (
+	"fmt"
+	"io"
+)
 
 type Consumer interface {
 	Consume(reader io.Reader, destPath string) error
 }
+
+func GetConsumer(consumerName string) (Consumer, error) {
+	switch consumerName {
+	case "file":
+		return &FileWriter{}, nil
+	case "tar":
+		return &TarExtractor{}, nil
+	case "null":
+		return &NullWriter{}, nil
+	default:
+		return nil, fmt.Errorf("unknown consumer: %s", consumerName)
+	}
+}
```

Comment:
I don't like this. Is there something better we can do? - I'd really like to have this functionality but I don't like the "GetXXX" style function. Not sure if there is something better we can do.